diff -dur atmelwlandriver.orig/Makefile.kernelv2.6 atmelwlandriver/Makefile.kernelv2.6
--- atmelwlandriver.orig/Makefile.kernelv2.6	2004-08-25 05:17:05.000000000 +0200
+++ atmelwlandriver/Makefile.kernelv2.6	2004-08-30 21:22:33.286984975 +0200
@@ -5,15 +5,13 @@
 SCRIPTSDIR=$(TOPDIR)/scripts
 INC=$(TOPDIR)/src/includes
 VERSION=3.4.0.2
-MINOR_NUMBER=$(shell uname -r | cut -d . -f 3 | cut -d - -f 1)
 ID:=$(shell id -u)
 tmpDIR:=/tmp/atm$(shell date +%S.%s)
 tmpDES:=$(tmpDIR)/atmelwlandriver
 LVNET = $(shell pwd)/src/apps/cmd_line
 WINTER = $(shell pwd)/src/apps/winter
 
-KERNEL_VERSION:=$(shell uname -r)
-KERNEL_SOURCE:=/lib/modules/$(KERNEL_VERSION)/build
+KERNEL_SOURCE:=/usr/src/linux
 PCMCIA_DES = $(DESTDIR)/lib/modules/$(KERNEL_VERSION)/pcmcia
 PCI_DES = $(DESTDIR)/lib/modules/$(KERNEL_VERSION)/kernel/drivers/net
 USB_DES = $(DESTDIR)/lib/modules/$(KERNEL_VERSION)/kernel/drivers/usb
@@ -99,34 +97,15 @@
 $(all_targets): %:
 	@echo Building $@
 	@if [ ! -d $(OBJDIR)/$@ ]; then echo Bootstraping target $@; mkdir $(OBJDIR)/$@; fi 
-	@if [ ! -d $(OBJDIR)/$@/debug ]; then \
-		mkdir $(OBJDIR)/$@/debug; \
-		for f in $(srcs); do (cd $(OBJDIR)/$@/debug; ln -s $(SUBDIRS)/$$f .); done \
-	fi
 	@if [ ! -d $(OBJDIR)/$@/release ]; then \
 		mkdir $(OBJDIR)/$@/release; \
 		for f in $(srcs); do (cd $(OBJDIR)/$@/release; ln -s $(SUBDIRS)/$$f .); done \
 	fi 
-	@echo -e "\tDebug"
-	@if [ "$(buildonly)" == debug ] || [ "$(buildonly)" == "" ]; then \
-		echo obj-m=$@.o > $(OBJDIR)/$@/debug/Makefile; \
-		echo all: >> $(OBJDIR)/$@/debug/Makefile; \
-		if [ "$(MINOR_NUMBER)" -gt "5" ]; then \
-			echo -e $(debug_submake_new) >> $(OBJDIR)/$@/debug/Makefile; \
-		else \
-			echo -e $(debug_submake_old) >> $(OBJDIR)/$@/debug/Makefile; \
-		fi;\
-		(cd $(OBJDIR)/$@/debug; make) ; \
-	fi
 	@echo -e "\tRelease"
 	@if [ "$(buildonly)" == release ] || [ "$(buildonly)" == "" ]; then \
 		echo obj-m=$@.o > $(OBJDIR)/$@/release/Makefile; \
 		echo all: >> $(OBJDIR)/$@/release/Makefile; \
-		if [ "$(MINOR_NUMBER)" -gt "5" ]; then \
-			echo -e $(release_submake_new) >> $(OBJDIR)/$@/release/Makefile; \
-		else \
-			echo -e $(release_submake_old) >> $(OBJDIR)/$@/release/Makefile; \
-		fi;\
+		echo -e $(release_submake_new) >> $(OBJDIR)/$@/release/Makefile; \
 		(cd $(OBJDIR)/$@/release; make) ; \
 	fi
 
diff -dur atmelwlandriver.orig/src/Pcmcia_Pci/Makefile atmelwlandriver/src/Pcmcia_Pci/Makefile
--- atmelwlandriver.orig/src/Pcmcia_Pci/Makefile	2004-08-25 05:17:04.000000000 +0200
+++ atmelwlandriver/src/Pcmcia_Pci/Makefile	2004-08-30 21:21:59.985610940 +0200
@@ -5,22 +5,7 @@
 include .lastbuild
 endif
 
-ARCH=$(shell uname -m)
-VR:=$(shell uname -r)
-ID:=$(shell id -u)
-ifneq ($(VR),$(UNAME))
-this:
-#	@echo -ne "\\033[1;31m"
-	@echo -ne "Running kernel doesn't match with the source"
-#	@echo -ne "\\033[0;39m"
-	@echo
-endif
-
-CC:=gcc
-CDEBUG:= -DPCMCIA_DEBUG=1
-DEFS :=  -fno-strict-aliasing -fno-common -pipe -mpreferred-stack-boundary=2 -march=$(ARCH) -fno-optimize-sibling-calls 
-CFLAGS_$(CONFIG_MODVERSIONS):= -DMODULE -DMODVERSIONS -include $(KERNEL_SRC)/include/linux/modversions.h
-CFLAGS := -DLINUX_OS -D__KERNEL__ -Wall -Wstrict-prototypes -Wno-trigraphs -O2 $(DEFS) -I$(KERNEL_SRC)/include -I$(INC) -I$(INC)/pcmcia $(CFLAGS_y)
+CFLAGS += -I$(INC) -I$(INC)/pcmcia $(CFLAGS_y)
 PCMCIAADDONFLAGS := -I$(PCMCIA_SRC)/include
 srcs := card.c command.c common.c fastvnet_cs.c interrupt.c mgmt.c rx.c tx.c vnet.c vnetlinux.c 
 pcisrcs := $(patsubst fastvnet_cs.c, fastvnet_cs.c ,$(srcs))
diff -dur atmelwlandriver.orig/src/apps/cmd_line/Makefile atmelwlandriver/src/apps/cmd_line/Makefile
--- atmelwlandriver.orig/src/apps/cmd_line/Makefile	2004-08-25 05:17:04.000000000 +0200
+++ atmelwlandriver/src/apps/cmd_line/Makefile	2004-08-30 21:21:59.988610523 +0200
@@ -1,22 +1,8 @@
-KERNEL_VERSION_NUMBER=$(shell uname -r | cut -d . -f 2)
-
-ifeq '$(KERNEL_VERSION_NUMBER)' '4'
-include $(TOPDIR)/.config
-endif
-
 OBJs=../../../objs
-FLAGS1:=-Wall -DATMEL_WLAN -DATMEL -O2 -g
+FLAGS1:=-Wall -DATMEL_WLAN -DATMEL $(OPT)
 FLAGS2:=-lncurses
-FLAGS3:=-DLOW_RES
-CC:=gcc
 
-ifeq '$(KERNEL_VERSION_NUMBER)' '4'
-all:
-	$(CC) $(FLAGS1) -I$(INC) -o lvnet $(FLAGS2) cofvnet.c sets.c survey.c 
-	@install -m 755 lvnet $(OBJs)/
-else
 INC = ../../includes
 all:
-	$(CC) $(FLAGS1) -I$(INC) -o lvnet $(FLAGS2) cofvnet.c sets.c survey.c 
+	$(CC) $(FLAGS1) -I$(INC) -I/usr/include/ncurses -o lvnet $(FLAGS2) cofvnet.c sets.c survey.c 
 	@install -m 755 lvnet $(OBJs)/
-endif
diff -dur atmelwlandriver.orig/src/apps/fw-upgrade/Makefile atmelwlandriver/src/apps/fw-upgrade/Makefile
--- atmelwlandriver.orig/src/apps/fw-upgrade/Makefile	2004-08-25 05:17:04.000000000 +0200
+++ atmelwlandriver/src/apps/fw-upgrade/Makefile	2004-08-30 21:21:59.988610523 +0200
@@ -7,7 +7,7 @@
 CCC:=gcc
 CFLAGS:=$(shell wx-config --cxxflags) 
 CFLAGS1:=$(shell wx-config --libs)
-CCFLAGS:= -O2 -Wall -Wstrict-prototypes
+CCFLAGS:=$(OPT)
 OUTDIR:="/usr/local/bin"
 
 
diff -dur atmelwlandriver.orig/src/usb/Makefile atmelwlandriver/src/usb/Makefile
--- atmelwlandriver.orig/src/usb/Makefile	2004-08-25 05:17:05.000000000 +0200
+++ atmelwlandriver/src/usb/Makefile	2004-08-30 21:21:59.987610662 +0200
@@ -1,13 +1,8 @@
 include .lastbuild
 include $(TOPDIR)/.config
-ARCH=$(shell uname -m)
 MODULE := usbvnet.o
-CDEBUG := -DUSBDBG
-DEFS := -fno-strict-aliasing -fno-common -pipe -mpreferred-stack-boundary=2 -march=$(ARCH) -fno-optimize-sibling-calls 
 USB_DES := $(MODULES_DES)/kernel/drivers/net
-CC=gcc
-CFLAGS_$(CONFIG_MODVERSIONS):= -DMODULE -DMODVERSIONS -include $(KERNEL_SRC)/include/linux/modversions.h
-CFLAGS :=-D__KERNEL__ -Wall -Wstrict-prototypes -Wno-trigraphs -O2 $(DEFS) -I$(KERNEL_SRC)/include -I$(INC) -I$(INC)/usb 
+CFLAGS += -I$(KERNEL_SRC)/drivers/usb/core -I$(INC) -I$(INC)/usb 
 
 $(CFLAGS_y)
 # Add extra compilation flags if under version 2.5 or higher
