--- atmelwlandriver/src/apps/winter/Makefile.rules	2006-04-11 20:05:40.000000000 +0200
+++ atmelwlandriver/src/apps/winter/Makefile.rules	2011-08-23 07:29:34.426583091 +0300
@@ -5,9 +5,9 @@
 
 all: postmake
 
-CC:=$(shell wx-config --cxx)
-CFLAGS:=$(shell wx-config --cxxflags) -I$(INC_DIR) -I$(RSRC_DIR) $(ARCH_SPECIFIC_CFLAGS)
-CFLAGS1:=$(shell wx-config --libs)
+CC:=$(shell $(WXCONFIG) --cxx)
+CFLAGS:=$(shell $(WXCONFIG) --cxxflags) -I$(INC_DIR) -I$(RSRC_DIR) $(ARCH_SPECIFIC_CFLAGS)
+CFLAGS1:=$(shell $(WXCONFIG) --libs)
 
 VPATH:=$(INC_DIR) $(SRC_DIR) $(RSRC_DIR)
 
@@ -20,13 +20,13 @@
 wxconfig:
 	@config_exists=no;\
 	for path_dir in `echo $(PATH) | tr : ' '`; do\
-	  if test -f $$path_dir/wx-config; then\
+	  if test -f $$path_dir/$(WXCONFIG); then\
 	    config_exists=yes;\
-	  fi\
+	  fi;\
 	done;\
 	if test $$config_exists = 'no'; then\
 	  echo 'It seems that the wxWindows library is not installed.';\
-	  echo 'wx-config, the script providing information about wxWindows';\
+	  echo '$(WXCONFIG), the script providing information about wxWindows';\
 	  echo 'installation, is missing. Please, refer to the README.linux';\
 	  echo 'or visit www.wxwindows.org for further information.';\
 	  echo 'We are sorry about the inconvenience.';\
