--- atmelwlandriver/src/usb/reset_device.c.orig	2004-07-18 23:50:44.000000000 +0200
+++ atmelwlandriver/src/usb/reset_device.c	2004-07-19 00:04:03.736081160 +0200
@@ -24,6 +24,9 @@
 #include <linux/delay.h>
 #include "linux/device.h"
 
+#define usb_snddefctrl(dev)	((PIPE_CONTROL << 30) | __create_pipe(dev,0))
+#define usb_rcvdefctrl(dev)	((PIPE_CONTROL << 30) | __create_pipe(dev,0) | USB_DIR_IN)
+
 static DECLARE_MUTEX(usb_address0_sem);
 
 static inline struct device *hubdev (struct usb_device *dev)
