--- atmelwlandriver/src/apps/fw-upgrade/Makefile.orig	2004-08-12 22:31:50.000000000 +0200
+++ atmelwlandriver/src/apps/fw-upgrade/Makefile	2004-08-12 22:34:26.000000000 +0200
@@ -3,10 +3,10 @@
 #########################################
 
 
-CC:=$(shell wx-config --cxx)
+CC:=$(shell wxgtk-2.4-config --cxx)
 CCC:=gcc
-CFLAGS:=$(shell wx-config --cxxflags) 
-CFLAGS1:=$(shell wx-config --libs)
+CFLAGS:=$(OPT) $(shell wxgtk-2.4-config --cxxflags) 
+CFLAGS1:=$(OPT) $(shell wxgtk-2.4-config --libs)
 CCFLAGS:=$(OPT)
 OUTDIR:="/usr/local/bin"
 
@@ -34,7 +34,7 @@
 wxconfig:
 	@config_exists=no;\
 	for path_dir in `echo $(PATH) | tr : ' '`; do\
-	  if test -f $$path_dir/wx-config; then\
+	  if test -f $$path_dir/wxgtk-2.4-config; then\
 	    config_exists=yes;\
 	  fi\
 	done;\
